utils/remote-ext-tests/replay-block/src/util.rs

```diff
@@ -0,0 +1,236 @@
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Helper functions for creating remote externalities, executor etc.
+
+use crate::{HashFor, NumberFor, LOG_TARGET};
+
+use frame_remote_externalities::{
+    Builder, Mode, OnlineConfig, RemoteExternalities, TestExternalities,
+};
+#[cfg(feature = "always-wasm")]
+use sc_executor::sp_wasm_interface::HostFunctions;
+#[cfg(not(feature = "always-wasm"))]
+use sc_executor::{NativeElseWasmExecutor, NativeExecutionDispatch};
+use sc_executor::{WasmExecutor, WasmtimeInstantiationStrategy};
+use sp_core::{
+    offchain::{
+        testing::{TestOffchainExt, TestTransactionPoolExt},
+        OffchainDbExt, OffchainWorkerExt, TransactionPoolExt,
+    },
+    storage::well_known_keys,
+    testing::TaskExecutor,
+    traits::{CallContext, CodeExecutor, TaskExecutorExt},
+    twox_128,
+};
+use sp_externalities::Extensions;
+use sp_keystore::{testing::KeyStore, KeystoreExt};
+use sp_rpc::{list::ListOrValue, number::NumberOrHex};
+use sp_runtime::{
+    generic::SignedBlock,
+    traits::{Block as BlockT, Header as HeaderT},
+    DeserializeOwned,
+};
+use sp_state_machine::{
+    backend::BackendRuntimeCode, ExecutionStrategy, OverlayedChanges, StateMachine,
+};
+use std::{fmt::Debug, str::FromStr, sync::Arc};
+use substrate_rpc_client::{ChainApi, WsClient};
+
+#[cfg(not(feature = "always-wasm"))]
+pub(crate) fn build_executor<D: NativeExecutionDispatch>() -> NativeElseWasmExecutor<D> {
+    let heap_pages = Some(2048);
+    let max_runtime_instances = 8;
+    let runtime_cache_size = 2;
+
+    NativeElseWasmExecutor::<D>::new_with_wasm_executor(WasmExecutor::new(
+        sc_executor::WasmExecutionMethod::Interpreted,
+        heap_pages,
+        max_runtime_instances,
+        None,
+        runtime_cache_size,
+    ))
+}
+
+#[cfg(feature = "always-wasm")]
+pub(crate) fn build_executor<H: HostFunctions>() -> WasmExecutor<H> {
+    let heap_pages = Some(2048);
+    let max_runtime_instances = 8;
+    let runtime_cache_size = 2;
+
+    WasmExecutor::new(
+        sc_executor::WasmExecutionMethod::Compiled {
+            instantiation_strategy: WasmtimeInstantiationStrategy::RecreateInstanceCopyOnWrite,
+        },
+        heap_pages,
+        max_runtime_instances,
+        None,
+        runtime_cache_size,
+    )
+}
+
+#[allow(dead_code)]
```

Comment:
deadcode?

Reply:
It's not used anymore, kept it in case we need something like this in the future (parse a string literal into a block hash). However, it's quite trivial anyway - I think it can be safely removed for now